lib/lambda_ethereum_consensus/engine/execution.ex

```diff
@@ -0,0 +1,26 @@
+defmodule LambdaEthereumConsensus.Engine.Execution do
+  alias LambdaEthereumConsensus.RPC
+
+  # Default Execution Layer endpoint
+  @execution_engine_endpoint "http://0.0.0.0:8551"
+
+  # Default Execution Layer RPC version
+  @execution_engine_rpc_version "2.0"
+
+  @spec engine_exchange_capabilities(list) :: {:ok, any} | {:error, any}
+  def engine_exchange_capabilities(capabilities) do
+    params = %{
+      "capabilities" => capabilities
+    }
+
+    case RPC.call(
+           "engine_exchangeCapabilities",
+           @execution_engine_endpoint,
+           @execution_engine_rpc_version,
+           params
+         ) do
+      {:ok, result} -> RPC.validate_response(result)
+      {:error, error} -> {:error, error}
+    end
```

Comment:
Today I learned that we can use [`with`](https://elixir-lang.org/getting-started/mix-otp/docs-tests-and-with.html#with) for mapping `:ok` results:

```suggestion
    with {:ok, result} <- RPC.call(
           "engine_exchangeCapabilities",
           @execution_engine_endpoint,
           @execution_engine_rpc_version,
           params
         ) do
      RPC.validate_response(result)
    end
```

Reply:
But then how do i manage the error case ?

Reply:
You can't use it for that, but in this case, you aren't actually handling the error but just propagating it upwards. With `with`, if the match fails, the non-matching result is returned, and so the behaviour is _mostly_ the same as with `case`.